StartProcess/handler.ts

```diff
@@ -0,0 +1,138 @@
+  ResponseSuccessAccepted
+} from "@pagopa/ts-commons/lib/responses";
+
+import { CommaSeparatedListOf } from "@pagopa/ts-commons/lib/comma-separated-list";
+
+import { flow, pipe } from "fp-ts/lib/function";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredQueryParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_query_param";
+import { withDefault } from "@pagopa/ts-commons/lib/types";
+import { SqlQuerySpec } from "@azure/cosmos";
+import { Context } from "@azure/functions";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { IMembership, IpaCode, MembershipStatus } from "../models/types";
+import { Dao } from "../models/dao";
+import { QueueItem } from "../ProcessMembership/handler";
+
+const composeQuery = (
+  ipas: ReadonlyArray<IpaCode>,
+  limit: number,
+  status: MembershipStatus
+): SqlQuerySpec => {
+  // when a list of IpaCodes is provides, we retrieve all of them
+  if (ipas.length) {
+    return {
+      parameters: [
+        { name: "@ipas", value: ipas },
+        { name: "@status", value: status }
+      ],
+      query:
+        "SELECT * FROM contracts d WHERE d.status = @status  d.ipaCode IN @ipaCode"
+    };
+  }
+  // otherwise we fetch the first elements on the selected status
+  else {
+    return {
+      parameters: [
+        { name: "@limit", value: limit },
+        { name: "@status", value: status }
+      ],
+      query:
+        "SELECT * FROM contracts d WHERE d.status = @status OFFSET 0 LIMIT @limit"
+    };
+  }
+};
+
+type MembershipWithFiscalCode = t.TypeOf<typeof MembershipWithFiscalCode>;
+const MembershipWithFiscalCode = t.intersection([
+  IMembership,
+  t.type({ fiscalCode: NonEmptyString })
+]);
+
+const prepareQueueItem = ({
+  ipaCode,
+  fiscalCode
+}: MembershipWithFiscalCode): QueueItem => ({
+  fiscalCode,
+  ipaCode
+});
+const dispatch = (context: Context) => (
+  items: ReadonlyArray<QueueItem>
+): ReadonlyArray<QueueItem> => {
+  // eslint-disable-next-line functional/immutable-data
+  context.bindings.processMembership = RA.map(JSON.stringify)(items);
+  return items;
+};
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+export function StartProcess({
+  dao
+}: {
+  readonly dao: Dao;
+}): express.RequestHandler {
+  const middlewaresWrap = withRequestMiddlewares(
+    ContextMiddleware(),
+    // list of ipa codes
+    //   use this parameter to select a specic set of memberships to process
+    RequiredQueryParamMiddleware(
+      "ipas",
+      withDefault(CommaSeparatedListOf(IpaCode), [])
+    ),
+    // limit the number of membership processed
+    //   applied only if ipas is not defined
+    RequiredQueryParamMiddleware("limit", withDefault(t.number, 100)),
```

Comment:
Why is limit required? When you want to processo all memberships there should be no limit

Reply:
I see it may be misleading, hence I tried to get explained it in the comments. _required+default_ means "you can leave it blank, we'll use a default value anyway". 

One preferable option would have been to be explicit about all the possible configurations we can have in the whole querystring hashmap:
```ts
{ ipas: NonEmptyArray<string> }
|| { limit: NonNegativeNumber }
```

I decided it wasn't worth the effort

Reply:
ok, so if we wanto to process all the memberships (aka with no limit) we have to set a "big" limit value (like 100000), correct?

Reply:
I see your point... Yes, it would work, I didn't actually consider such scenario to be honest